backend/api/routes/archives.py

```diff
@@ -0,0 +1,143 @@
+    languages: list[str] | None
+    tags: list[str] | None
+
+
+class ArchiveRequest(BaseModel):
+    filename: str | None
+    email: str | None
+    config: ArchiveConfig
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class ArchiveModel(BaseModel):
+    id: UUID
+
+    project_id: UUID
+
+    filename: str
+    filesize: int
+    created_on: datetime.datetime
+    download_url: str
+    status: str
+    email: str | None
+    config: dict[str, Any]
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+def validated_archive(
+    archive_id: UUID,
+    project: Project = Depends(validated_project),
+    session: Session = Depends(gen_session),
+) -> Archive:
+    """Depends()-able archive from request, ensuring it exists"""
+    stmt = select(Archive).filter_by(id=archive_id).filter_by(project_id=project.id)
+    archive = session.execute(stmt).scalar()
+    if not archive:
+        raise HTTPException(HTTPStatus.NOT_FOUND, f"Archive not found: {archive_id}")
+    return archive
+
+
+@router.get("/{project_id}/archives", response_model=list[ArchiveModel])
+async def get_all_archives(
+    project: Project = Depends(validated_project),
+) -> list[ArchiveModel]:
+    """Get all archives of a project"""
+    return TypeAdapter(list[ArchiveModel]).validate_python(project.archives)
+
+
+@router.get("/{project_id}/archives/{archive_id}", response_model=ArchiveModel)
+async def get_archive(archive: Archive = Depends(validated_archive)) -> ArchiveModel:
+    """Get a specific archives of a project"""
+    return ArchiveModel.model_validate(archive)
+
+
+@router.post(
+    "/{project_id}/archives",
+    response_model=ArchiveModel,
+    status_code=HTTPStatus.CREATED,
+)
+async def create_archive(
+    archive_request: ArchiveRequest,
+    project: Project = Depends(validated_project),
+    session: Session = Depends(gen_session),
+) -> ArchiveModel:
+    """Create a pre-request archive"""
+    now = datetime.datetime.now(tz=datetime.UTC)
+    new_archive = Archive(
+        filename=archive_request.filename
+        if archive_request.filename
+        else f"{project.name}.zim",
+        filesize=project.used_space,
+        created_on=now,
+        requested_on=datetime.datetime.min,
+        download_url="",
+        collection_json_path="",
+        status=ArchiveStatus.CREATED,
+        zimfarm_task_id=constants.empty_uuid,
+        email=archive_request.email,
+        config=archive_request.config.model_dump(),
+    )
+    project.archives.append(new_archive)
+    session.add(new_archive)
+    session.flush()
+    session.refresh(new_archive)
+    return ArchiveModel.model_validate(new_archive)
+
+
+@router.patch(
+    "/{project_id}/archives/{archive_id}",
+    status_code=HTTPStatus.NO_CONTENT,
+)
+async def update_archive(
+    archive_request: ArchiveRequest,
+    archive: Archive = Depends(validated_archive),
+    project: Project = Depends(validated_project),
+    session: Session = Depends(gen_session),
+):
```

Comment:
For now, I haven't add a support for Illustration image. I think I should, But I don't know what's the best way to deal with this part.

Reply:
Yes ZIM Illustration is important. You may use a dedicated endpoint if you think that makes sense UX-wise.

It's a regular file upload for which you:
- check that its declared as a std image
- check uploaded filesize (fail if too big)
- convert to PNG using scraperlib
- resize to 48x48 using scraperlib
- store as base64. That's appropriate for our small 48px PNG. We could store as bytea but we'd have to serialize for sending and we don't have much use of this in the server.

In the UI, the process will be mirrored:
- check with JS the filetype and filesize (constants to config the max illustration size –– 2MiB?)
- upload ; alert if it failed, display if OK

Reply:
Sorry for late reply, new semester really catch me. 😢
I will create a separate endpoint for this.